src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java

```diff
@@ -371,6 +375,13 @@ void testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecksForUpda
         assertEquals("OE111229", filing.getData().get("entityNumber"));
         assertEquals("OE02", filing.getData().get("type"));
 
+        final OverseasEntitySubmissionDto submissionDto = (OverseasEntitySubmissionDto)filing.getData().get("userSubmission");
```

Comment:
I know this is following the pattern already established in the file, but we shouldn't be doing an unchecked cast.

Also same with others below the variable type isn't necessary here and it's usually preferred to use the Local Variable Type Inference (`var`) where possible.

Reply:
```suggestion
        assertInstanceOf(OverseasEntitySubmissionDto.class, filing.getData().get("userSubmission"));
        final var submissionDto = (OverseasEntitySubmissionDto)filing.getData().get("userSubmission");
```